src/Summoner/Question.hs

```diff
@@ -63,9 +63,8 @@ queryDef question defAnswer = do
     if | T.null answer -> pure defAnswer
        | otherwise     -> pure answer
 
-queryManyRepeatOnFail :: forall a . (Text -> Maybe a) -> Text -> IO [a]
-queryManyRepeatOnFail parser question = do
-    T.putStrLn question
+queryManyRepeatOnFail :: forall a . (Text -> Maybe a) -> IO [a]
+queryManyRepeatOnFail parser = do
```

Comment:
You don't need `do` here and you can put `promptLoop` on the same line as `queryManyRepeatOnFail`